lib/itoolkit.js

```diff
@@ -569,7 +569,29 @@ class iSql {
   }
 }
 
-
+exports.types = {
+    char(countChars) {
+        return countChars.toString() + "A";
+    },
+    packed(digits, delimiter) {
+        return digits.toString() + "P" + delimiter.toString();
+    },
+    zoned(digits, delimiter) {
+        return digits.toString() + "S" + delimiter.toString();
+    },
+    signed(digits, delimiter) {
+        return digits.toString() + "I" + delimiter.toString();
+    },
+    unsigned(digits, delimiter) {
+        return digits.toString() + "U" + delimiter.toString();
+    },
+    float(digits, delimiter) {
+        return digits.toString() + "F" + delimiter.toString();

```

Comment:
I would rather see separate `float` and `double` functions which take no parameters and return "4F2" and "8F4", respectively.

Aside: I'm not sure why XMLSERVICE wants 4F2 and 8F4, when RPG says the decimal digits field must be blank (and would be 4F and 8F, respectively).